core/src/main/java/io/cloudevents/core/message/impl/MessageUtils.java

```diff
@@ -59,6 +59,15 @@ public static MessageReader parseStructuredOrBinaryMessage(
             EventFormat format = EventFormatProvider.getInstance().resolveFormat(ct);
             if (format != null) {
                 return structuredMessageFactory.apply(format);
+            } else {
+                /**
+                 * The format wasn't one we support, but if it's part of the
+                 * CloudEvent family it indicates it's a structured
+                 * representation that we can't interpret.
+                 */
+                if (ct.startsWith("application/cloudevents")) {
```

Comment:
I see what you're trying to do here. AFAIK this logic is nowhere in the cloudevents sdks. But I see it's following this wording: https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#3-http-message-mapping

Reply:
It might be worth updating the other sdks too